Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -127,36 +130,72 @@ public class PushDeploymentController : ApiController
             }
         }
 
-        private async Task<HttpResponseMessage> PushDeployAsync(ZipDeploymentInfo deploymentInfo, bool isAsync)
+        private async Task<string> GetZipURLFromJSON()
         {
-            if (_settings.RunFromLocalZip())
+            // ARM template should have properties field and a packageUri field inside the properties field.
+            var requestObject = await Request.Content.ReadAsAsync<JObject>();
+            using (_tracer.Step("Reading the zip URL from the request JSON"))
             {
-                await WriteSitePackageZip(deploymentInfo, _tracer);
+                try
+                {
+                    return requestObject.Value<JObject>("properties").Value<string>("packageUri");
+                }
+                catch (Exception ex)
+                {
+                    _tracer.TraceError(ex, "Error reading the URL from the JSON");
```

Comment:
Maybe also log `requestObject.ToString()` so if you needed to go back and look at it you know what the request content actually was.